services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerEndpoint.java

```diff
@@ -434,19 +517,37 @@ public Response getCredentialOfferPreflight(@PathParam("sessionCode") String ses
      */
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    @Path(CREDENTIAL_OFFER_PATH + "{sessionCode}")
-    public Response getCredentialOffer(@PathParam("sessionCode") String sessionCode) {
+    @Path(CREDENTIAL_OFFER_PATH + "{nonce}")
+    public Response getCredentialOffer(@PathParam("nonce") String nonce) {
         configureCors(false);
 
-        if (sessionCode == null) {
+        if (nonce == null) {
             throw new BadRequestException(getErrorResponse(ErrorType.INVALID_CREDENTIAL_REQUEST));
         }
 
-        CredentialsOffer credentialsOffer = getOfferFromSessionCode(sessionCode);
-        LOGGER.debugf("Responding with offer: %s", credentialsOffer);
+        RealmModel realm = session.getContext().getRealm();
+        EventBuilder eventBuilder = new EventBuilder(realm, session, session.getContext().getConnection());
 
-        return cors.add(Response.ok()
-                .entity(credentialsOffer));
+        var offerStorage = session.getProvider(CredentialOfferStorage.class);
+        var offerEntry = offerStorage.findOfferStateByNonce(nonce);
+        if (offerEntry == null) {
+            var errorMessage = "No offer entry for nonce: " + nonce;
+            eventBuilder.event(INTROSPECT_TOKEN_ERROR).detail(Details.REASON, errorMessage).error(Errors.INVALID_REQUEST);
+            throw new BadRequestException(getErrorResponse(ErrorType.INVALID_TOKEN));
+        }
+
+        var credOffer = offerEntry.credentialsOffer;
```

Comment:
I am not sure how the other developers think. But I actually like strong typing and not using the `var`-keyword.  It blurs the readability of the code in my opinion.